Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestParticipationValidationService.cs

```diff
@@ -82,4 +83,55 @@ public ValidationResult GetValidationResult((Contest?, int?, UserInfoModel?, boo
 
         return ValidationResult.Valid();
     }
+
+    public ValidationResult GetValidationResult((ContestServiceModel?, UserInfoModel?, bool) item)
+    {
+        var (contest, user, official) = item;
+
+        // TODO: Fix so it uses lecturers in contests business service
+        var userIsAdminOrLecturerInContest = user != null && (user.IsAdmin || this.lecturersInContestsBusiness
+            .IsCurrentUserAdminOrLecturerInContest(contest?.Id)
+            .GetAwaiter()
+            .GetResult());
```

Comment:
Why not refactor the existing method, instead of adding new one? Is it used on many places?

Reply:
Fixed.